examples/twitter-beam-java/src/main/java/TweeterStream.java

```diff
@@ -0,0 +1,35 @@
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.transforms.windowing.*;
+import org.apache.beam.sdk.values.PCollection;
+import org.joda.time.Duration;
+
+import java.util.Collections;
+
+public class TweeterStream {
+
+    public static void main(String[] args) {
+        Pipeline pipeline = Pipeline.create();
+        PCollection<String> tweetStream = pipeline.apply(Create.of(Collections.singletonList(""))).apply(ParDo.of(new ReadFromTweeterDoFn()))
```

Comment:
For sources it's usually more convenient to provide an interface e.g. `TwitterIO` that takes a configuration. Perhaps something like this:


For providing the configuration:
```
pipeline.apply(TwitterIO.readTweets()
    .withUserToken(...)
    .with...())
```

And for using a properties file, perhaps
```
pipeline.apply(TwitterIO.withConfigFile('twitter4j.properties'))
```

What do you think?

You would override PTransform, and in the `expand` transform you would do the whole `pipeline.apply(Create.of(Collections.singletonList(""))).apply(ParDo.of(new ReadFromTweeterDoFn()))`, so that the Create and the ReadFrom.... are just implementation details

Reply:
Tried to do this, please review.